test/test_autograd.py

```diff
@@ -4669,22 +4669,39 @@ def test_grad_fn_attr_bindings(self):
         out = torch.stack([a, b], dim=0)
         self.assertEqual(out.grad_fn._saved_tensors, (a, b))              # TensorList -> Tuple[Tensor]
         self.assertIsInstance(out.grad_fn._saved_tensors[0], torch.Tensor)
+        self.assertIsInstance(out.grad_fn._raw_saved_tensors[0], torch._C._autograd.SavedTensor)
         self.assertEqual(out.grad_fn._saved_dim, 0)                       # int64_t -> int
         self.assertIsInstance(out.grad_fn._saved_dim, int)
 
+        saved = out.grad_fn._raw_saved_tensors[0]
+        saved.register_hooks(lambda x: x, lambda x: x)
+
         out.sum().backward()
         with self.assertRaisesRegex(RuntimeError, "after they have already been freed"):
             out.grad_fn._saved_tensors
+        with self.assertRaisesRegex(RuntimeError, "after they have already been freed"):
+            out.grad_fn._raw_saved_tensors
+        with self.assertRaisesRegex(RuntimeError, "after it has been freed"):
+            saved.register_hooks(lambda x: x, lambda x: x)
         self.assertEqual(out.grad_fn._saved_dim, 0)
 
         a = torch.ones(2, 2, requires_grad=True)
         indices = torch.tensor([0, 1])
         out = a[:, indices]
         self.assertEqual(out.grad_fn._saved_indices, (None, indices))     # c10::List<c10::optional<Tensor>> -> Tuple[Tensor?]
         self.assertIsInstance(out.grad_fn._saved_indices[1], torch.Tensor)
+        self.assertIsInstance(out.grad_fn._raw_saved_indices[1], torch._C._autograd.SavedTensor)
         self.assertEqual(out.grad_fn._saved_self_sizes, a.shape)          # IntArrayRef -> Tuple[int]
         self.assertIsInstance(out.grad_fn._saved_self_sizes[0], int)
 
+        saved = out.grad_fn._raw_saved_indices
+        saved[1].register_hooks(lambda x: x, lambda x: x)
+        with self.assertRaisesRegex(RuntimeError, "None is forbidden"):
+            saved[0].register_hooks(lambda x: x, lambda x: x)
+        out.sum().backward()
+        with self.assertRaisesRegex(RuntimeError, "after it has been freed"):
+            saved[1].register_hooks(lambda x: x, lambda x: x)
```

Comment:
same for here

Reply:
we should probably just not do this test, and instead add a comment above the `del saved` on the lifetime issue

Reply:
My plan is to remove this test for cases 2 and 3 of the quip I sent you privately (so whenever we deal with a  `std::vector<SavedVariable>`) but keep it in case 1 because we're not actually deleting the `SavedVariable` during backward (only calling `.reset_data()` on it).